people/views.py

```diff
@@ -70,6 +70,28 @@ def profile(request, year, pk):
 		'roles': RecruitmentApplication.objects.filter(user = user, status = 'accepted').order_by('recruitment_period__fair')
 	})
 
+def profile_delete(request, year, pk):
+	fair = get_object_or_404(Fair, year = year)
+	user = get_object_or_404(User, pk = pk)
+
+	profile = Profile.objects.filter(user = user).first()
+	if not profile: profile = Profile.objects.create(user = user, no_dietary_restrictions = False)
+	
+	application = RecruitmentApplication.objects.filter(user = user, status = 'accepted', recruitment_period__fair = fair).first()
+	
+
+	if request.method == 'POST':
+		Profile.objects.filter(user = request.user).delete()
+		user.delete()
+		return redirect('/accounts/logout?next=/register')
```

Comment:
This if-statement could happen earlier in the code after row 75 as you only waiting for a `request of POST` and then you delete the profile and user which have nothing to do with the code above.